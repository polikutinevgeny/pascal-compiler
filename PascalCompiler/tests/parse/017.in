const
   a1 = 30;
   b = a1*20;
var
   a: integer;
   function foo(): integer;
   begin

   end;

begin
   a := foo();
   begin
      if a > 20 then
         a := -a1;
   end;
end.
