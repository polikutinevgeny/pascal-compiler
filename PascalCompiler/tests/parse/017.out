program 
  integer: (integer = (built-in));
  real: (real = (built-in));
  char: (char = (built-in));
  a1: a1 :  = 30;
  b: b :  = 600;
  a: a : (integer = (built-in)) ;
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  function foo(), return type : (integer = (built-in))
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
├─Assignment statement
│ ├─Designator
│ │ └─a
│ └─Call in expression
│   └─Designator
│     └─foo
└─compound statement
  └─If
    ├─Greater
    │ ├─Designator
    │ │ └─a
    │ └─20
    └─Assignment statement
      ├─Designator
      │ └─a
      └─Minus
        └─Designator
          └─a1
